server.go

```diff
@@ -20,9 +20,18 @@ func receiveCodeViaLocalServer(ctx context.Context, c *Config) (string, error) {
 	if err != nil {
 		return "", xerrors.Errorf("error while starting a local server: %w", err)
 	}
+	if c.TLSConfig != nil {
+		if c.TLSConfig.CertFile == "" {
+			return "", fmt.Errorf("certificate must be set when using TLS serving")
+		}
+		if c.TLSConfig.KeyFile == "" {
+			return "", fmt.Errorf("certificate key must be set when using TLS serving")
+		}
+		listener.URL.Scheme = "https"
+	}
 	defer listener.Close()
```

Comment:
`listener.Close()` should be before the above block to close the listener even if error occurs.